Update DTLS cipher list

kadoshita opened the pull request:
Fix #956 

The current DTLS cipher suite setting is `HIGH:!CAMELLIA:!aNULL` , in which case the Client Hello fragmentation occurs.
Therefore, if a library that does not support the Client Hello fragmentation, such as Mbed TLS, is used, the handshake will fail.

This problem can be solved by aligning the cipher suite to the browser and reducing the number of cipher suites.

|<img width="1077" alt="before" src="https://github.com/aiortc/aiortc/assets/18162391/527ece9e-62bd-42fa-bfa1-a30fac9ece2b">|<img width="953" alt="after" src="https://github.com/aiortc/aiortc/assets/18162391/17a31d58-ab79-4360-8147-35f810b2198b">|
|:--:|:--:|
|before|after|

In my environment, the number of cipher suites sent by Client Hello is 12, as shown below, and I checked that fragmentation does not occur.

lgrahl commented:
[RFC 8827 states that](https://datatracker.ietf.org/doc/html/rfc8827#section-6.5) only `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256` with P-256 is mandatory and this aligns with what I've seen in the wild. Considering this and that aiortc does not like large RSA certificates (#828), I think we should also exclude all RSA cipher suites.

kadoshita commented:
Thank you.

I added `:!aRSA` and changed it to exclude RSA.
The cipher suites are as follows after changed.

```
Cipher Suites (5 suites)
    Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 (0xc02b)
    Cipher Suite: TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 (0xcca9)
    Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA (0xc009)
    Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA (0xc00a)
    Cipher Suite: TLS_EMPTY_RENEGOTIATION_INFO_SCSV (0x00ff)
```

jlaine commented:
Hi! I'm a bit confused by the string we are passing: since we're explicitly settings some cipher suites, why do we also need to exclude RSA?

kadoshita commented:
For example, AES128-SHA uses RSA as the key exchange algorithm and authentication algorithm.
The Google Chrome cipher suite included AES128-SHA, so we followed it.
However, I received a comment that cipher suites that use RSA should be excluded, so I updated it to exclude RSA.

Instead of excluding RSA, should we exclude cipher suites that use RSA as a key exchange or authentication algorithm?

jlaine commented on src/aiortc/rtcdtlstransport.py:
```diff
@@ -152,7 +152,7 @@ def _create_ssl_context(self) -> SSL.Context:
         ctx.use_certificate(self._cert)
         ctx.use_privatekey(self._key)
-        ctx.set_cipher_list(b"HIGH:!CAMELLIA:!aNULL")
+        ctx.set_cipher_list(b"ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES128-SHA:AES256-SHA:!aRSA")
```
```suggestion
        ctx.set_cipher_list(b"ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA:AES128-GCM-SHA256:AES128-SHA:AES256-SHA")
```

jlaine commented:
My point was: why declare RSA-based cipher suites then disable them with `!aRSA`, why not simply pull them from the list?

kadoshita commented:
Sorry.
I think it would be better to exclude RSA-based cipher suites from the list, as you suggested.

jlaine commented:
I've squashed down your commits, rebased on top of `main` and made the commit a bit more descriptive.

@fippo @lgrahl could you please comment on whether this cipher list is OK so we can get this merged?

kadoshita commented:
@jlaine @fippo @lgrahl Thank you for your reviewing!
